client/apps/edit/components/layout/index.coffee

```diff
@@ -34,7 +35,12 @@ module.exports = class EditLayout extends Backbone.View
 
   setupOnBeforeUnload: ->
     window.onbeforeunload = =>
-      if $.active > 0 then "Your article is not finished saving." else null
+      if $.active > 0
+        "Your article is not finished saving."
+      else if @changedAPublishedArticle is true and not @finished
```

Comment:
I wonder if you could use `@article.changed` instead here http://backbonejs.org/#Model-changed. Not a blocker though.